libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+    if (!_initialised) {
+        return false;
+    }
+
+    FrameType frame_to_send;
+
+    switch (tracking_type) {
+    case TrackingType::TRK_NONE:{
+        bool disable_track = false;
+        frame_to_send = {
+        .flag={0},
+        data_len : sizeof(disable_track),
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_track_switch),
+        data : (uint8_t*)&disable_track
+        };
+        // send packet to gimbal
+        return send_packet(frame_to_send);
+    }
+    case TrackingType::TRK_POINT: {
+        uint16_t pos[2] = {uint16_t(p1.x * 10000),uint16_t(p1.y * 10000)};
+        bool start_track = true;
+        if (pos[0] == 5000 && pos[1] == 5000) {
+            // set deadtime over 500ms to avoid trigger tracking mistake
+            if (AP_HAL::millis() - _last_tracking_ms < 500) {
+                return 0;
+            }
+            frame_to_send = {
+            .flag={0},
+            data_len : sizeof(start_track),
+            cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+            sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_track_switch),
+            data : (uint8_t*)&start_track
+            };
+
+            _last_tracking_ms = AP_HAL::millis();
+        } else {
+            frame_to_send = {
+                .flag={0},
+                data_len : sizeof(pos),
+                cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+                sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_trackpos_set),
+                data : (uint8_t*)pos
+            };
+        }
+
+        return send_packet(frame_to_send);
+    }
+    case TrackingType::TRK_RECTANGLE:{
+
+        uint8_t track_window[2] = {uint8_t((p2.x-p1.x)*255),uint8_t((p2.y-p1.y)*255)};
+        frame_to_send = {
+        .flag={0},
+        data_len : sizeof(track_window),
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_trackbox_set),
+        data : (uint8_t*)track_window
+        };
+
+        send_packet(frame_to_send);
+
+        uint16_t pos[2] = {uint16_t((p2.x - p1.x) * 10000/2),uint16_t((p2.y - p1.y) * 10000/2)};
+        
+        frame_to_send = {
+            .flag={0},
+            data_len : sizeof(pos),
+            cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+            sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_trackpos_set),
+            data : (uint8_t*)pos
+        };
+        return send_packet(frame_to_send);
+    }
+    
+    default:{
+        return 0;
+    }
+    }
+}
+
+// set camera lens as a value from 0 to 3
+bool AP_Mount_Tusuav::set_lens(uint8_t lens) 
+{
+    // exit immediately if not initialised
+    if (!_initialised) {
+        return false;
+    }
+
+    if (lens > (uint8_t)CameraSource::PIP_IR) {
+        return false;
+    }
+
+    MessageCameraSource camera = {
```

Comment:
```suggestion
    const MessageCameraSource camera{
```

... any many other places

Reply:
Ok,thank you！